micro_sam/training/training.py

```diff
@@ -93,11 +92,13 @@ def train_sam(
     freeze: Optional[List[str]] = None,
     device: Optional[Union[str, torch.device]] = None,
     lr: float = 1e-5,
-    scheduler: Optional[_LRScheduler] = None,
     n_sub_iteration: int = 8,
     save_root: Optional[Union[str, os.PathLike]] = None,
     mask_prob: float = 0.5,
     n_iterations: Optional[int] = None,
+    scheduler_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
It's correct to remove the scheduler since it needs the optimizer, but it would be nice to enable another type of scheduler. We could introduce another argument `scheduler_class`, which defaults to `ScheduleLROnPlateau`, but makes the choice of scheduler fully flexible.

Reply:
I agree, makes sense to expose it as well. Updated it.